src/backend/distributed/executor/multi_utility.c

```diff
@@ -335,6 +342,11 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 	if (IsA(parsetree, TruncateStmt))
 	{
 		ErrorIfUnsupportedTruncateStmt((TruncateStmt *) parsetree);
+
+		if (ShouldExecuteTruncateStmtSequential((TruncateStmt *) parsetree))
```

Comment:
I think this logic should live in `master_modify_multiple_shards` because we're already doing something related there and if we put it here the logic would be scattered:

```C
f (ShouldExecuteTruncateStmtSequential((TruncateStmt *) parsetree))
{
	SetLocalMultiShardModifyModeToSequential();
}

... and the code continues with
if (MultiShardConnectionType == SEQUENTIAL_CONNECTION)
{

}